### PR TITLE
fix: generateAddressFromPubKey and tests

### PR DESCRIPTION
**Pull request by ieow**
Removes a lot of unused and redundant code.
Fixes private key generation.
Fixes generating Ethereum addresses.
Properly serializes public key using secp256k1 library instead of helpers.
Sorts json keys for signature verification on server.
Removes unnecessary complexity from tests.
Updates test values from torus;.js
Adds missing else case where enableOneKey is true and typeOfUser is v1.
Consolidates secp256k1 implementations.
etc.

### CODE REVIEW

**Comment by matthiasgeihs** [comment]
Closes #62

File: Sources/TorusUtils/Models/RetrieveSharesResponseModel.swift [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
+    public let publicAddress: String
+    public let privateKey: String
 
-     public init(publicKey: String, privateKey: String) {
-         self.publicAddress = publicKey
-         self.privateKey = privateKey
-     }
- }
+    public init(publicKey: String, privateKey: String) {
+        publicAddress = publicKey
```
Comment (ieow) [inline Sources/TorusUtils/Models/RetrieveSharesResponseModel.swift:8]: why remove `self` ?
if remove `self` has valid reason, why privateKey's `self` below is not removed

Comment (metalurgical) [reply Sources/TorusUtils/Models/RetrieveSharesResponseModel.swift:8]: `self` only needs to be explicitly stated here if the variable names and the function parameters conflict.

`privateKey` (assigned with value of parameter `privateKey`) here needs an explicit self, `publicAddress` (assigned with value of parameter `publicKey`) does not.

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift [hunk 2]
```diff
@@ -91,181 +80,143 @@ extension TorusUtils {
...
 
-    
-    
     // MARK: - getShareOrKeyAssign
-    
+
     private func getShareOrKeyAssign(endpoints: [String], nodeSigs: [CommitmentRequestResponse], verifier: String, verifierParams: VerifierParams, idToken: String, extraParams: [String: Any] = [:]) async throws -> [URLRequest] {
-        let session = createURLSession()
-        let threshold =  Int(endpoints.count / 2) + 1
+        _ = createURLSession()
```
Comment (ieow) [inline Sources/TorusUtils/Extensions/TorusUtils+extension.swift:144]: removed this ?

Comment (metalurgical) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:144]: The value was never used, just assigned to a variable here.

**Review by ieow — COMMENTED** [review]
lgtm

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 2 above)
Comment (ieow) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:144]: yea, I mean should we remove these?

**Review by metalurgical — APPROVED** [review]

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

**Pull request merged**